crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -133,6 +155,236 @@ fn is_fast_corner(
+    let p7 = get_pixel_from_offset(6);
+    let p8 = get_pixel_from_offset(7);
+    let p10 = get_pixel_from_offset(9);
+    let p11 = get_pixel_from_offset(10);
+    let p12 = get_pixel_from_offset(11);
+    let p14 = get_pixel_from_offset(13);
+    let p15 = get_pixel_from_offset(14);
+    let p16 = get_pixel_from_offset(15);
+    let pixels = [
+        p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15, p16,
+        p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15, p16 
+    ]; // Values are repeated to handle circular indexing
+    
+    let mut consecutive_brighter = 0u8;
+    let mut consecutive_darker = 0u8;
+
+    let mut is_corner = false;
+    let mut start_idx = 0;
+    let mut end_idx = 0;
+
+    let mut most_consecutive = 0u8;
+    let update_indices = |consecutive: &u8, idx: &usize, start_idx: &mut usize, end_idx: &mut usize, most_consecutive: &mut u8| {
+        if consecutive > most_consecutive {
+            *end_idx = (idx+1).rem_euclid(16);
+            *most_consecutive = *consecutive;
+            *start_idx = ((*end_idx + 16) - *most_consecutive as usize).rem_euclid(16);
+        }
+    };
+    
+    for (idx, &pixel) in pixels.iter().enumerate() {
+        // Iterate through all pixels. Only update start and end idx when consecutive count increases.
+        if pixel > upper_threshold {
+            consecutive_brighter += 1;
+            consecutive_darker = 0;
+            
+            // If we were counting brighter pixels...
+            update_indices(&consecutive_brighter, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+        } else if pixel < lower_threshold {
+            consecutive_darker += 1;
+            consecutive_brighter = 0;
+            
+            // If we were counting darker pixels...
+            update_indices(&consecutive_darker, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+        } else {
+            update_indices(&consecutive_brighter, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            update_indices(&consecutive_darker, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+
+            consecutive_brighter = 0;
+            consecutive_darker = 0;
+        }
+
+        if consecutive_brighter >= arc_length || consecutive_darker >= arc_length {
+            is_corner = true;
+        }
+
+        if consecutive_brighter >= 16 {
+            update_indices(&consecutive_brighter, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            break;
+        } else if consecutive_darker >= 16 {
+            update_indices(&consecutive_darker, &idx, &mut start_idx, &mut end_idx, &mut most_consecutive);
+            break;
+        }
+    }
+
+    if !is_corner {
+        return (false, 0); // Not a corner found
+    }
+
+    // Sum of absolute differences for the corner score. 
+    let mut score = 0i32;
+    for offset in start_idx..start_idx + most_consecutive as usize {
```

Comment:
This probably can be done while check the arc ? To avoid this extra loop

Reply:
Agreed.